LinkedListImpl.java

```diff
@@ -77,6 +77,25 @@ public E getFirst() {
         return firstElement != null ? firstElement.value: null;
     }
 
+    @Override
+    public Iterator<E> iterator() {
+        Iterator<E> iter = new Iterator<E>() {
+            @Override
+            public boolean hasNext() {
+                return firstElement != null;
+            }
+
+            @Override
+            public E next() {
+                E result = firstElement.value;
+                if(firstElement == null){
+                    throw new IndexOutOfBoundsException("First Element is null");
+                }
```

Comment:
проверку надо выполнять до, иначе получим NPE
```suggestion
                if(firstElement == null){
                    throw new IndexOutOfBoundsException("First Element is null");
                }
                E result = firstElement.value;
```